src/lib/parsers/sol-json.c

```diff
@@ -1092,6 +1092,10 @@ sol_json_load_memdesc(const struct sol_json_token *token, const struct sol_memde
             len++;
         }
 
+        /* never allocate zero bytes */
+        if (len == 0)
+            return -EINVAL;
```

Comment:
i think it's ok. `len == 0` will happen due a not correct `desc`, that is a user input. For those cases we are always returning `-EINVAL`